app/uk/gov/hmrc/economiccrimelevyregistration/navigation/ConfirmContactAddressPageNavigator.scala

```diff
@@ -38,10 +38,20 @@ class ConfirmContactAddressPageNavigator @Inject() (eclRegistrationConnector: Ec
         eclRegistrationConnector
           .upsertRegistration(registration = registration.copy(contactAddress = registration.grsAddressToEclAddress))
           .map(_ => routes.CheckYourAnswersController.onPageLoad())
-      case Some(false) => Future.successful(routes.IsUkAddressController.onPageLoad())
+      case Some(false) => Future.successful(routes.IsUkAddressController.onPageLoad(NormalMode))
       case _           => Future.successful(routes.JourneyRecoveryController.onPageLoad())
     }
 
-  override protected def navigateInCheckMode(registration: Registration): Future[Call] = ???
+  override protected def navigateInCheckMode(registration: Registration)(implicit
+    request: RequestHeader
+  ): Future[Call] =
+    registration.useRegisteredOfficeAddressAsContactAddress match {
```

Comment:
This is a duplicate of the normal mode routing behaviour (but only because this is the final page in the journey if the registered address is used)... do we want to put the navigation logic into a common function?